Code/Tools/FBuild/FBuildTest/Data/TestExec/execenv.cpp

```diff
@@ -0,0 +1,30 @@
+//
+// An simple executable to run
+//
+#include <iostream>
+#include <fstream>
+#include <string>
```

Comment:
FASTBuild doesn't use C++ standard libraries at all. It uses C standard libraries only.

This will make this test significantly slower to compile, but also introduces problems with C++ runtime libs (which even when statically linked have various issues on all platforms)

I think this test could be simplified to check that retrieving the env vars are the exact ones we know the test sets and returning 0 or 1 as a return value. This would avoid the need for fstream, iostream and string (removing all c++ standard library use)

Core/Env/Env.cpp has environment variable access code that is free of C++ standard library use that could be used as a reference.